### PR TITLE
docs/commitlog: remove mastership election

### PR DESCRIPTION
**Pull request by daviddrysdale**
The commitlog simulation code didn't previously make clear -- it should work without mastership election.  The simkafka STHs topic effectively acts as the serialization point: N signers attempt to write a new STH at offset x, but only one of those updates actually gets written to offset x, and that's the one that counts.

### CODE REVIEW

File: docs/storage/commit_log/main.go [hunk 1]
```diff
@@ -34,6 +35,7 @@ import (
 )
 
 var (
+	runElections        = flag.Bool("run_elections", false, "Whether to run mastership election; if false, signers run in parallel")
```
Comment (AlCutter) [inline docs/storage/commit_log/main.go]: "run" vs "use"?

File: docs/storage/commit_log/main.go [hunk 2]
```diff
@@ -68,28 +70,51 @@ func increment(s string) string {
 	return string(append([]byte(increment(prefix)), 'A'))
 }
 
+type atomicBool struct {
```
Comment (AlCutter) [inline docs/storage/commit_log/main.go]: BTW, there's `sync/atomic` which provides atomic ints, you could implement this in terms of those with `CompareAndSwap()` and save a few lines perhaps?

File: docs/storage/commit_log/main.go (hunk 1 above)
Comment (daviddrysdale) [reply docs/storage/commit_log/main.go]: Done.

File: docs/storage/commit_log/main.go (hunk 2 above)
Comment (daviddrysdale) [reply docs/storage/commit_log/main.go]: Atomics are best avoided; renamed to `lockedBool` to reduce the temptation in future.

**Comment by daviddrysdale** [comment]
@Bren2010 might be interested in this: I've changed the emphasis of the simulation code from "works even if mastership election is bugged" to "doesn't use a master election".  

The writes to the STHs topic give a serialization -- N signers attempt to write to offset+1, but only one of them actually writes there and N-1 write to a higher offset.  This is kind-of analogous to having N parallel database transactions where only one of them succeeds in committing and N-1 end up being rolled back.

**Pull request merged**